cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -2611,6 +2611,106 @@ TEST_F(ScalarTemporalTest, TestRoundTemporal) {
   CheckScalarUnary(op, unit, times, unit, round_15_years, &round_to_15_years);
 }
 
+TEST_F(ScalarTemporalTest, TestCeilFloorRoundTemporalAmbiguous1) {
+  auto unit = timestamp(TimeUnit::MILLI, "Asia/Tehran");
+  auto options = RoundTemporalOptions(1, CalendarUnit::HOUR, true);
+  const char* times = R"([
```

Comment:
It's a bit late to suggest this, but I think we should take the habit of making tests more easily understood by spelling out the context, e.g.:
```suggestion
  // Asia/Tehran switched from UTC+X to UTC+Y on 2022-03-31 HH:mm:ss
  const char* times = R"([
```